code/controllers/subsystem/shuttles/supply.dm

```diff
@@ -16,34 +16,53 @@
 	height = 7
 	roundstart_move = "supply_away"
 
+	var/list/blacklist = list(
+		/mob/living,
+		/obj/effect/blob,
+		/obj/effect/rune,
+		/obj/effect/spider/spiderling,
+		/obj/item/weapon/disk/nuclear,
+		/obj/machinery/nuclearbomb,
+		/obj/item/device/radio/beacon,
+		/obj/machinery/the_singularitygen,
+		/obj/singularity,
+		/obj/machinery/teleport/station,
+		/obj/machinery/teleport/hub,
+		/obj/machinery/telepad,
+		/obj/machinery/clonepod
+	)
+
 /obj/docking_port/mobile/supply/New()
 	..()
 	SSshuttle.supply = src
 
 /obj/docking_port/mobile/supply/canMove()
 	if(z == ZLEVEL_STATION)
-		return forbidden_atoms_check(areaInstance)
+		return check_blacklist(areaInstance)
 	return ..()
 
-/obj/docking_port/mobile/supply/request(obj/docking_port/stationary/S)
+/obj/docking_port/mobile/supply/proc/check_blacklist(atom/A)
+	if(is_type_in_list(A, blacklist))
+		return 1
+	for(var/thing in A)
+		if(.(thing))
```

Comment:
This was existing code, I just moved it around. Agreed though, a better solution would be nice.